cyber/service_discovery/container/multi_value_warehouse.cc

```diff
@@ -61,23 +61,19 @@ void MultiValueWarehouse::Remove(uint64_t key) {
 void MultiValueWarehouse::Remove(uint64_t key, const RolePtr& role) {
   WriteLockGuard<AtomicRWLock> lock(rw_lock_);
   auto range = roles_.equal_range(key);
-  for (auto it = range.first; it != range.second;) {
+  for (auto it = range.first; it != range.second; ++it) {
     if (it->second->Match(role->attributes())) {
       it = roles_.erase(it);
-    } else {
-      ++it;
```

Comment:
Changed logic. This is a common pattern of `loop and erase some items from a map`.